lib/chef/provider/package/yum.rb

```diff
@@ -1059,16 +1077,13 @@ def load_current_resource
             end
           end
 
-          # Don't overwrite an existing arch
-          unless arch
-            parse_arch
-          end
 
           @current_resource = Chef::Resource::Package.new(@new_resource.name)
           @current_resource.package_name(@new_resource.package_name)
 
           installed_version = []
           @candidate_version = []
+          @arch = []
```

Comment:
No, this is a class variable in the provider, but that the user passes in is stored in the new resource object (@new_resource.arch). We only overwrite that one in the one special case below, which is preserved from the original code.